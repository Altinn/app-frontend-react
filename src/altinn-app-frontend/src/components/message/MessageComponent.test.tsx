import * as React from 'react';
import { render } from '@testing-library/react';

import { MessageComponent, MessageType } from './MessageComponent';

describe('MessageComponent', () => {
  const mockId = 'mockId';
  let mockMessageType: MessageType;
  let mockMessages: string[];

  beforeEach(() => {
    mockMessageType = 'message';
    mockMessages = ['this is a message'];
  });
  it('should match snapshot', () => {
    const { container } = render(
      <MessageComponent
        messageType={mockMessageType}
        style={{ display: 'block', width: 'fit-content' }}
        key={'messageType'}
        id={mockId}
      >
        <ol>
          {mockMessages.map((message: string, idx: number) => {
            return <li key={idx}>{message}</li>;
          })}
        </ol>
      </MessageComponent>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
  it('should have class a-message-info when into type', () => {
    mockMessageType = 'info';
    const { container } = render(
      <MessageComponent
        messageType={mockMessageType}
        style={{ display: 'block', width: 'fit-content' }}
        key={'messageType'}
        id={mockId}
      >
        <ol>
          {mockMessages.map((message: string, idx: number) => {
            return <li key={idx}>{message}</li>;
          })}
        </ol>
      </MessageComponent>,
    );
    expect(
      container.querySelector('#mockId.a-message-info'),
    ).toBeInTheDocument();
  });
  it('should have class a-message-error when error type', () => {
    mockMessageType = 'error';
    const { container } = render(
      <MessageComponent
        messageType={mockMessageType}
        style={{ display: 'block', width: 'fit-content' }}
        key={'messageType'}
        id={mockId}
      >
        <ol>
          {mockMessages.map((message: string, idx: number) => {
            return <li key={idx}>{message}</li>;
          })}
        </ol>
      </MessageComponent>,
    );
    expect(
      container.querySelector('#mockId.a-message-error'),
    ).toBeInTheDocument();
  });
  it('should have class a-message-success when success type', () => {
    mockMessageType = 'success';
    const { container } = render(
      <MessageComponent
        messageType={mockMessageType}
        style={{ display: 'block', width: 'fit-content' }}
        key={'messageType'}
        id={mockId}
      >
        <ol>
          {mockMessages.map((message: string, idx: number) => {
            return <li key={idx}>{message}</li>;
          })}
        </ol>
      </MessageComponent>,
    );
    expect(
      container.querySelector('#mockId.a-message-success'),
    ).toBeInTheDocument();
  });
});
