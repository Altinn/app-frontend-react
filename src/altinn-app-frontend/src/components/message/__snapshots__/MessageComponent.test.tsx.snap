// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageComponent should match snapshot 1`] = `
<div
  class="field-validation-error a-message"
  id="mockId"
  style="display: block;"
>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
    >
      <i
        class=""
        style="margin-top: -2px; font-size: 1.8em;"
      />
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10"
    >
      <ol>
        <li>
          this is a message
        </li>
      </ol>
    </div>
  </div>
</div>
`;

